script/FullDeployer.s.sol

```
@@ -38,13 +47,33 @@ contract FullDeployer is HubDeployer, ExtendedSpokeDeployer {
         uint16 centrifugeId;
         string memory environment;
         string memory network;
+        address rootAddress;
 
         try vm.envString("NETWORK") returns (string memory _network) {
             network = _network;
             string memory configFile = string.concat("env/", network, ".json");
             string memory config = vm.readFile(configFile);
             centrifugeId = uint16(vm.parseJsonUint(config, "$.network.centrifugeId"));
             environment = vm.parseJsonString(config, "$.network.environment");
+            
+            try vm.parseJsonAddress(config, "$.network.root") returns (address _root) {
+                rootAddress = _root;
+            } catch {
+                rootAddress = address(0);
+            }
+            
+            // Parse messageCost and maxBatchSize with defaults
+            try vm.parseJsonUint(config, "$.network.messageCost") returns (uint256 _messageCost) {
+                messageCost = _messageCost;
+            } catch {
+                messageCost = 1_000_000; // 1M gas
+            }
+            
+            try vm.parseJsonUint(config, "$.network.maxBatchSize") returns (uint256 _maxBatchSize) {
+                maxBatchSize = _maxBatchSize;
+            } catch {
+                maxBatchSize = 50_000_000; // 50M gas
+            }
```

I think we should panic here if the value is not configured. The defaults were previously used for tests.